includes/class-simple-local-avatars.php

```diff
@@ -1222,7 +1233,44 @@ private function clear_user_avatar_cache( $local_avatars, $user_id, $media_id )
 	}
 
 	/**
-	 * Migrate the user's avatar data from WP User Avatar/ProfilePress
+	 * Add default avatar upload file field.
+	 *
+	 * @param array $defaults Default options for avatar.
+	 *
+	 * @return array Default options of avatar.
+	 */
+	public function add_avatar_default_field( $defaults ) {
+		if ( ! did_action( 'wp_enqueue_media' ) ) {
+			wp_enqueue_media();
+		}
+		$default_avatar_file_id  = get_option( 'simple_local_avatar_default', '' );
+		$default_avatar_file_url = wp_get_attachment_image_url( $default_avatar_file_id );
+		ob_start();
+		?>
+		<input type="hidden" name="simple-local-avatar-file-id" id="simple-local-avatar-file-id" value="<?php echo esc_attr( $default_avatar_file_id ); ?>"/>
+		<input type="hidden" name="simple-local-avatar-file-url" id="simple-local-avatar-file-url" value="<?php echo esc_url( $default_avatar_file_url ); ?>"/>
+		<input type="button" name="simple-local-avatar" id="simple-local-avatar-upload" class="button-secondary" value="<?php esc_html_e( 'Choose Default Avatar', 'simple-local-avatar' ); ?>"/>
+		<?php
+		$defaults['simple_local_avatar'] = ob_get_clean();
+
+		return $defaults;
+	}
+
+	/**
+	 * Save default avatar attachment id in option.
+	 */
+	private function save_default_avatar_file_id() {
+		global $pagenow;
+
+		$file_id = filter_input( INPUT_POST, 'simple-local-avatar-file-id', FILTER_SANITIZE_NUMBER_INT );
+
+		// check for uploaded files
+		if ( 'options.php' === $pagenow && ! empty( $file_id ) ) {
```

Comment:
I wonder if `$file_id` could be `empty` as [filter_input](https://www.php.net/manual/en/function.filter-input.php) returns `false` if the filter fails, or `null` if the variable is not set.

Reply:
@faisal-alvi we have put a check for empty here. if null or empty, it will not update the option. You are suggesting we should put some validation message for user if it fails?